stac_fastapi/pgstac/core.py

```diff
@@ -341,6 +342,14 @@ async def post_search(
             ItemCollection containing items which match the search criteria.
         """
         item_collection = await self._search_base(search_request, request=request)
+
+        # If we have the `fields` extension enabled
+        # we need to avoid Pydantic validation because the
+        # Items might not be a valid STAC Item objects
+        if fields := getattr(search_request, "fields", None):
+            if fields.include or fields.exclude:
+                return JSONResponse(item_collection)  # type: ignore
```

Comment:
https://stac-utils.github.io/stac-fastapi/migrations/v3.0.0/#fields-extension-and-model-validation